File: spectra-backend/models/spot.py
from typing import Optional

from pydantic import BaseModel


class Spot(BaseModel):
    id: str
    name: str
    latitude: float
    longitude: float
    region: str
    nearest_buoy_id: Optional[str] = None
    nearest_wind_station: Optional[str] = None
    nearest_tide_station: Optional[str] = None

File: Spectra/Models/Spot.swift
import Foundation

struct Spot: Identifiable, Codable {
    let id: UUID
    let name: String
    let latitude: Double
    let longitude: Double
    let region: String
    let nearestBuoyId: String?
    let nearestWindStation: String?
    let nearestTideStation: String?
}
